app/views/user_mailer/_gpx_description.html.erb

```diff
@@ -1,12 +1,8 @@
-<%= t "user_mailer.gpx_notification.your_gpx_file" %>
-<strong><%= @trace_name %></strong>
-<%= t "user_mailer.gpx_notification.with_description" %>
-<em><%= @trace_description %></em>
-<% if @trace_tags.length>0 %>
-  <%= t "user_mailer.gpx_notification.and_the_tags" %>
-  <em><% @trace_tags.each do |tag| %>
-    <%= tag.tag.rstrip %>
-  <% end %></em>
+<% trace_name = tag.strong(@trace_name) %>
+<% trace_description = tag.em(@trace_description) %>
+<% if @trace_tags.length > 0 %>
+  <% tags = @trace_tags.map { tag.tag.rstrip.join(" ") } %>
```

Comment:
Did you mean to put that `join` after the `map` rather than inside it? Also should it be `safe_join` or is interpolating it into an `_html` resource going to have much the same effect>

Reply:
Two good catches; the join should be after the map, and I'll look into the `safe_join` aspect too. And maybe add some tests :smile:

Reply:
> and I'll look into the `safe_join` aspect too

Actually, I don't think we want the safe_join here, since `tag.tag` isn't made up of existing safe strings (e.g. already-built html) - `tag.tag` is just a regular string. So `join` is the best approach and the whole string will be properly escaped when interpolated into the erb.

Of course, `join` is still in the wrong place!